client/app/components/file-uploader/component.js

```diff
@@ -10,15 +10,24 @@ export default Ember.Component.extend({
             const reader = new FileReader();
             const file_handle = ev.target.files[0];
             const saveParameter = this.attrs.saveParameter
+            const filename_parts = ev.currentTarget.value.split('\\')
+            const filename = filename_parts[filename_parts.length - 1];
+            const parameters = this.attrs.widget.value.parameters;
+
+            saveParameter(parameters.fileName, {
+                value: filename,
+                state: ['defined']
+            });
 
             reader.onloadend = function(ev) {
-                saveParameter({
+                saveParameter(parameters.fileData, {
```

Comment:
What if there are multiple of the same component? Will it override the data from the first component?